src/pages/index.js

```diff
@@ -119,45 +181,129 @@ function closeModalEscape(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitButton = evt.submitter
+  submitButton.textContent = "Saving...."
+
+  api
+  .editUserInfo({ name: editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+  .then((userData) => {
+
+    profileName.textContent = userData.name;
+    profileDescription.textContent = userData.about;
+    closeModal(editModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+   submitButton.textContent = "Save";''
+
+  });
+
 }
 
-function handleAddCardSubmit(evt) {
+function handleSubmit(evt) {
   evt.preventDefault();
   const inputValues = {
     name: cardNameInput.value,
     link: cardLinkInput.value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(addCardModal);
+  const submitButton = evt.submitter;
+  SetButtonText(true, submitButton);
+
+  api
+  .addCard(inputValues)
+  .then((newCard) => {
+    const newCardElement = getCardElement(inputValues);
+    cardsList.prepend(newCardElement);
+    evt.target.reset();
+    disableButton(cardSubmitButton, validationConfig);
+    closeModal(addCardModal);
+
+  })
+  .finally(() => {
+    SetButtonText(false, submitButton);
+
+  });
+
+
+}
+
+
+function handleAvatarSubmit(evt)  {
+  evt.preventDefault();
```

Comment:
* According to the task, you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it. 


When I change the avatar or delete a card, the button doesn't change the text